### ISSUE TITLE
CS0029 since .NET SDK 5.0.200 when using ternary operator with FormattableString

### ISSUE THREAD

**Issue by m-gasser** [dotnet/roslyn#51674]
**Version Used**: 
.NET SDK 5.0.200

**Steps to Reproduce**:
1. Create a Program.cs with the following line of code
```cs
System.FormattableString f = bool.Parse("true") ? $"{int.MinValue}" : $"{int.MaxValue}";
```

**Expected Behavior**:
No compiler error

**Actual Behavior**:
[CS0029] Cannot implicitly convert type 'string' to 'System.FormattableString'

This seems to be either a bug or an undocumented breaking change in .NET SDK 5.0.200. It works wit .NET SDK 5.0.100 and also with of .NET 6 Preview 1.

**Comment by Youssef1313** [comment]
@m-gasser This unfortunately was accidentally allowed in .NET 5.0.103 and VS 16.8, and the current behavior is the expected. See https://github.com/dotnet/roslyn/issues/49598.

@jaredpar Wondering if this should be documented?

**Comment by m-gasser** [comment]
To get the code to compile again, we added a cast to one of the ternary branches. At least with JetBrains Rider the cast gets grayed out because it thinks it is unnecessary. I guess JetBrains has an eye on the breaking changes list so this is another reason why I think it is a good idea to document this.

**Comment by jaredpar** [comment]
Closing as this behavior is "By Design". It's a fix for a bug we introduced in 5.1.103. 

This will get added to our breaking changes doc. Generally that doc is reserved for breaking intended compiler behavior. In this case though it's fixing a short term bug. It's notable enough though that we should include it in the doc.

**Comment by cston** [comment]
Documented in [Compiler Breaking Changes - post DotNet 5.md](https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/Compiler%20Breaking%20Changes%20-%20post%20DotNet%205.md) in https://github.com/dotnet/roslyn/pull/51754.
